### DISCUSSION TITLE
It seems like $$ command doesn't work correctly since v8.35.0 release

### DISCUSSION THREAD

**Discussion by Nikola-Dz** [webdriverio/webdriverio#12544] [Technical Discussion]
Hi,

I have several tests where I fetch multiple elements with $$ command and then check for each element if it is displayed or not. Something like this:
```javascript
let elements = $$('.some-class')

for (let i = 0; i < await elements.length; i++) {
    
    if (condition 1) {
        await expect(elements[0]).toBeDisplayedInViewport();
    }
    
    else if (condition 2) {
        await expect(elements[0]).not.toBeDisplayed();
    }
}
``` 

Until the release of [v8.35.00 ](https://github.com/webdriverio/webdriverio/releases/tag/v8.35.0) - #12530 it worked perfectly fine, since then I'm getting an error. Now, if I fetch elements who are not displayed or not existing I'm getting this error:

```powershell
Error: Index out of bounds! $$(.some-class) returned only 0 elements.    
    at Object.executeCommandBe (file:///D:/Projekte%20D/WORK/Player-automation_-_WebdriverIO/player-automation/node_modules/expect-webdriverio/lib/utils.js:52:16)
    at Object.toBeDisplayed (file:///D:/Projekte%20D/WORK/Player-automation_-_WebdriverIO/player-automation/node_modules/expect-webdriverio/lib/matchers/element/toBeDisplayed.js:9:20)
``` 

Is this a bug or is there a workaround for this problem?

Help is much appreciated. 😀

**Comment by christian-bromann** [comment]
@Nikola-Dz mind raising an issue with a minimal reproducible example?

**Reply by Nikola-Dz** [reply]
I've added the issue here https://github.com/webdriverio/webdriverio/issues/12576

Thanks for your reply
